source/code/plugins/oms_common.rb

```diff
@@ -732,6 +732,11 @@ def create_ods_request(path, record, compress, extra_headers=nil, serializer=met
           headers["Content-Encoding"] = "deflate"
         end
 
+        headers["User-Agent"] = "LinuxMonitoringAgent/#{OMS::Common.get_agent_version}"
+        headers[OMS::CaseSensitiveString.new("x-ms-app")] = "LinuxMonitoringAgent"
+        headers[OMS::CaseSensitiveString.new("x-ms-client-version")] = OMS::Common.get_agent_version
```

Comment:
Could get_agent_version call fail/return empty

Reply:
Then we have to fix the get_agent_version itself.

Reply:
In some case it could be nil, at least should do a check like preexisting headers here against that to avoid sending a nil field

Reply:
nil field will be converted to empty string and the results will be User-Agent="LinuxMonitoringAgent/"

I guess that is fine, or I can make it as User-Agent="LinuxMonitoringAgent/0.0.0-0"

Reply:
I think the latter is more useful / sensible.